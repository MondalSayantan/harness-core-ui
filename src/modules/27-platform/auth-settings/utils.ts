/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get } from 'lodash-es'
import type { CreateIpAllowlistConfigProps, UpdateIpAllowlistConfigProps } from '@harnessio/react-ng-manager-client'

import type { IIPAllowlistForm } from '@auth-settings/interfaces/IPAllowlistInterface'
import { SourceType } from '@auth-settings/interfaces/IPAllowlistInterface'
import { CurrencyType } from '@common/constants/SubscriptionTypes'

export function getAmountInCurrency(currency: CurrencyType, amount: number): string {
  switch (currency) {
    case CurrencyType.USD: {
      return amount.toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD'
      })
    }
  }

  return ''
}

// dividing the actual amount by 100 as amount numbers are in cents
export function getDollarAmount(amount?: number, isYearly?: boolean): number {
  if (isYearly) {
    return (amount || 0) / 1200
  }
  return (amount || 0) / 100
}

export const getErrorMessage = (error: any): string =>
  get(error, 'data.error', get(error, 'data.message', error?.message))

export const buildCreateIPAllowlistPayload = (data: IIPAllowlistForm): CreateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: false, // always false while creating
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags
      }
    }
  }
}

export const buildUpdateIPAllowlistPayload = (
  data: IIPAllowlistForm,
  overrideFields?: any
): UpdateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    'ip-config-identifier': data.identifier,
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: data?.enabled,
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags,
        ...overrideFields
      }
    }
  }
}

/**
 * Converts minutes to higher dimensions of time such as months, days and hours
 *
 * Examples:
 * formatMinutesToHigherDimensions(150);    // Output: "2 hours 30 minutes"
 * formatMinutesToHigherDimensions(4320);   // Output: "3 days"
 * formatMinutesToHigherDimensions(45000);  // Output: "1 month 1 day 6 hours"
 * @param minutes
 * @returns a string in higher dimensions
 */
export const formatMinutesToHigherDimensions = (minutes: number | undefined): string => {
  if (minutes === undefined || minutes < 0) {
    return 'Invalid input'
  }

  const MINUTES_IN_AN_HOUR = 60
  const HOURS_IN_A_DAY = 24
  const DAYS_IN_A_MONTH = 30

  const result = []

  if (minutes >= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY * DAYS_IN_A_MONTH) {
    const months = Math.floor(minutes / (MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY * DAYS_IN_A_MONTH))
    result.push(`${months} month${months > 1 ? 's' : ''}`)
    minutes %= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY * DAYS_IN_A_MONTH
  }

  if (minutes >= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY) {
    const days = Math.floor(minutes / (MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY))
    result.push(`${days} day${days > 1 ? 's' : ''}`)
    minutes %= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY
  }

  if (minutes >= MINUTES_IN_AN_HOUR) {
    const hours = Math.floor(minutes / MINUTES_IN_AN_HOUR)
    result.push(`${hours} hour${hours > 1 ? 's' : ''}`)
    minutes %= MINUTES_IN_AN_HOUR
  }

  if (minutes > 0) {
    result.push(`${minutes} minute${minutes > 1 ? 's' : ''}`)
  }

  return result.join(' ')
}
