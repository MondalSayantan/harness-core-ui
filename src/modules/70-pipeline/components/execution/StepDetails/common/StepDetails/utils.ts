/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ResponseApprovalInstanceResponse } from 'services/pipeline-ng'

export function encodeURIWithReservedChars(uri: string): string {
  return encodeURIComponent(uri).replace(/[;,/?:@&=+$#]/g, function (c) {
    return '%' + c.charCodeAt(0).toString(16)
  })
}

export interface ExecutionMetadataType {
  approvalInstanceId: string
  mock?: {
    data?: ResponseApprovalInstanceResponse
    loading?: boolean
  }
}
