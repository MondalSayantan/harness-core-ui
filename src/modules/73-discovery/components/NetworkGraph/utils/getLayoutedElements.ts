/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import ELK, { ElkExtendedEdge, ElkNode } from 'elkjs/lib/elk.bundled'
import type { Edge } from 'reactflow'
import type { LayoutedGraph } from '../types'
import { NodeTypes } from '../constants'

const options = {
  'elk.algorithm': 'stress'
}

export async function getLayoutedElements({ nodes, edges }: LayoutedGraph): Promise<LayoutedGraph> {
  // Generate elk graph node from reactflow node structure
  // Get the group nodes
  const groupNodes = nodes.filter(node => node.type === NodeTypes.Group)

  const graph: ElkNode = {
    id: 'root',
    layoutOptions: options,
    // Create parent elk nodes with it's children and edges
    children: groupNodes.map(groupNode => {
      return {
        ...groupNode,
        // Hardcode a width and height for elk to use when layouting.
        width: 0,
        height: 0,
        children: nodes
          .filter(node => node.parentNode === groupNode.id)
          .map(node => ({
            ...node,
            // Adjust the target and source handle positions based on the layout
            // direction.
            targetPosition: 'top',
            sourcePosition: 'bottom',

            // Hardcode a width and height for elk to use when layouting.
            width: 100,
            height: 100
          })),
        edges: edges.filter(
          edge => /* istanbul ignore next */ edge.data?.parentNode === groupNode.id
        ) as unknown as ElkExtendedEdge[]
      }
    }),
    edges: edges as unknown as ElkExtendedEdge[]
  }

  const elk = new ELK()
  const elkGraph = await elk.layout(graph)

  /* istanbul ignore next */
  const flattenedNodes = elkGraph?.children?.map(node => ({
    ...node,
    // React Flow expects a position property on the node instead of `x`
    // and `y` fields.
    position: { x: node.x, y: node.y },
    style: {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      ...(node as any).style,
      width: node.width,
      height: node.height
    }
  }))

  /* istanbul ignore next */
  elkGraph?.children?.map(node => {
    if (node.children) {
      /* istanbul ignore next */
      flattenedNodes?.push(
        ...node.children.map(n => ({
          ...n,
          // React Flow expects a position property on the node instead of `x`
          // and `y` fields.
          position: { x: n.x, y: n.y },
          style: {}
        }))
      )
    }
  })

  const layoutedGraph = {
    nodes: flattenedNodes,
    edges: elkGraph.edges as unknown as Edge<unknown>[]
  } as LayoutedGraph

  return layoutedGraph
}
