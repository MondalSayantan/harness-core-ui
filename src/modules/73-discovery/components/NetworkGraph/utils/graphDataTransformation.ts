/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { Node, Edge } from 'reactflow'
import { NodeTypes, nodeGroupOptions, nodeOptions } from '@discovery/components/NetworkGraph/constants'
import type {
  ApiCreateNetworkMapRequest,
  ApiListCustomServiceConnection,
  ApiListK8sCustomService
} from 'services/servicediscovery'
import type {
  NetworkMapEdgeData,
  NetworkMapNodeData,
  ServiceEdgeData,
  ServiceNodeData
} from '@discovery/components/NetworkGraph/types'

export function getGraphNodesFromServiceList(
  serviceList: ApiListK8sCustomService | null
): Node<ServiceNodeData, NodeTypes>[] {
  if (!serviceList) return []

  const namespaces = new Set<string>()
  const graphNodes: Node<ServiceNodeData, NodeTypes>[] = []

  serviceList.items?.map(service => {
    if (service.namespace && !namespaces.has(service.namespace)) {
      namespaces.add(service.namespace)
    }

    if (service.id)
      graphNodes.push({
        id: service.id,
        data: service,
        parentNode: service.namespace,
        ...nodeOptions
      })
  })

  namespaces.forEach(value => {
    graphNodes.push({
      id: value,
      data: { name: value },
      ...nodeGroupOptions
    })
  })

  return graphNodes
}

export function getGraphEdgesFromServiceConnections(
  connectionList: ApiListCustomServiceConnection | null
): Edge<ServiceEdgeData>[] {
  if (!connectionList) return []

  const graphEdges: Edge<ServiceEdgeData>[] = []

  connectionList.items?.map(connection => {
    if (connection.id && connection.sourceID && connection.destinationID)
      graphEdges.push({
        id: connection.id,
        source: connection.sourceID,
        target: connection.destinationID,
        data: { parentNode: connection.sourceNamespace, ...connection }
      })
  })

  return graphEdges
}

export function getGraphNodesFromNetworkMap(
  networkMap: ApiCreateNetworkMapRequest | undefined
): Node<NetworkMapNodeData, NodeTypes>[] {
  if (!networkMap) return []

  const namespaces = new Set<string>()
  const graphNodes: Node<NetworkMapNodeData, NodeTypes>[] = []

  networkMap.resources?.map(service => {
    if (service.namespace && !namespaces.has(service.namespace)) {
      namespaces.add(service.namespace)
    }

    if (service.id)
      graphNodes.push({
        id: service.id,
        data: service,
        parentNode: service.namespace,
        ...nodeOptions
      })
  })

  namespaces.forEach(value => {
    graphNodes.push({
      id: value,
      data: { name: value },
      ...nodeGroupOptions
    })
  })

  return graphNodes
}

export function getGraphEdgesFromNetworkMap(
  networkMap: ApiCreateNetworkMapRequest | undefined
): Edge<NetworkMapEdgeData>[] {
  if (!networkMap) return []

  const graphEdges: Edge<NetworkMapEdgeData>[] = []

  networkMap.connections?.map(connection => {
    if (connection.from?.id && connection.to?.id)
      graphEdges.push({
        id: `${connection.from.id}-${connection.to.id}`,
        source: connection.from.id,
        target: connection.to.id,
        data: { parentNode: connection.from.namespace, ...connection }
      })
  })

  return graphEdges
}
