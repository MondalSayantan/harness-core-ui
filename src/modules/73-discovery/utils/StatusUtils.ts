/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Color } from '@harness/design-system'
import type { DatabaseDelegateTaskStatus } from 'services/servicediscovery'

export enum DAgentTaskStatus {
  SUCCESS = 'SUCCESS',
  FAILED = 'FAILED',
  ERROR = 'ERROR',
  PROCESSED = 'PROCESSED'
}

export const useDAgentStatusColorValue = (status?: DatabaseDelegateTaskStatus): string => {
  switch (status) {
    case DAgentTaskStatus.SUCCESS:
      return Color.GREEN_50
    case DAgentTaskStatus.FAILED:
      return Color.RED_50
    case DAgentTaskStatus.ERROR:
      return Color.ORANGE_50
    case DAgentTaskStatus.PROCESSED:
      return Color.YELLOW_100
    default:
      return Color.GREY_50
  }
}

export const useDAgentStatusTextColor = (status?: DatabaseDelegateTaskStatus): string => {
  switch (status) {
    case DAgentTaskStatus.SUCCESS:
      return Color.GREEN_700
    case DAgentTaskStatus.FAILED:
      return Color.RED_700
    case DAgentTaskStatus.ERROR:
      return Color.ORANGE_700
    case DAgentTaskStatus.PROCESSED:
      return Color.YELLOW_700
    default:
      return Color.GREY_700
  }
}
