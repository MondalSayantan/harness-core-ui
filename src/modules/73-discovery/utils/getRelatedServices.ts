/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import {
  ApiListCustomServiceConnection,
  ApiListK8sCustomService,
  DatabaseK8SCustomServiceCollection
} from 'services/servicediscovery'

export function getRelatedServices(
  serviceID: string,
  serviceList: ApiListK8sCustomService | null,
  connectionList: ApiListCustomServiceConnection | null
): DatabaseK8SCustomServiceCollection[] | undefined {
  if (!serviceList || !connectionList) return

  const relatedServiceIds = connectionList.items?.map(conn => {
    if (serviceID === conn.sourceID) return conn.destinationID ?? ''
    /* istanbul ignore next */
    if (serviceID === conn.destinationID) return conn.sourceID ?? ''
  })

  return serviceList?.items?.filter(service => service.id === serviceID || relatedServiceIds?.includes(service.id))
}
