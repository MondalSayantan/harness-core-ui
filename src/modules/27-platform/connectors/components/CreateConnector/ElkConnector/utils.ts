/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormData } from '@platform/connectors/interfaces/ConnectorInterface'
import { AuthTypes } from '@platform/connectors/pages/connectors/utils/ConnectorHelper'
import { UseStringsReturn } from 'framework/strings'
import type { ConnectorConfigDTO } from 'services/cd-ng'
import { ElkConnectorAuthFieldsProps } from './CreateElkConnector.types'
import { ELKConnectorFields, ElkAuthType } from './CreateElkConnector.constants'

type InitializeElkConnectorArgs = {
  prevStepData?: ConnectorConfigDTO
  accountId: string
  projectIdentifier?: string
  orgIdentifier?: string
}

export function initializeElkConnector({
  prevStepData,
  accountId,
  projectIdentifier,
  orgIdentifier
}: InitializeElkConnectorArgs): FormData {
  const defaultObj = {
    url: '',
    accountId,
    username: '',
    authType: AuthTypes.USER_PASSWORD,
    password: undefined,
    projectIdentifier,
    orgIdentifier
  }

  if (!prevStepData) {
    return defaultObj
  }
  const { spec, ...prevData } = prevStepData
  return {
    ...defaultObj,
    url: prevData?.url || spec?.controllerUrl || spec?.url || '',
    password: prevData?.password || spec?.passwordRef,
    clientSecretRef: prevData?.clientSecretRef || spec?.clientSecretRef,
    authType: prevData?.authType || spec?.authType || AuthTypes.USER_PASSWORD,
    username: prevData?.username || spec?.username || '',
    clientId: prevData?.clientId || spec?.clientId,
    apiKeyId: prevData?.apiKeyId || spec?.apiKeyId,
    apiKeyRef: prevData?.apiKeyRef || spec?.apiKeyRef
  }
}

interface GetElkAuthTypeReturn {
  label: string
  value: string
}

export const getELKAuthType = (getString: UseStringsReturn['getString']): GetElkAuthTypeReturn[] => [
  {
    label: getString('usernamePassword'),
    value: AuthTypes.USER_PASSWORD
  },
  {
    label: getString('common.apikey'),
    value: AuthTypes.API_CLIENT_TOKEN
  },
  {
    label: getString('platform.connectors.bearerToken'),
    value: AuthTypes.BEARER_TOKEN
  },
  {
    label: getString('platform.connectors.elk.noAuthentication'),
    value: ElkAuthType.NONE
  }
]

export const getAuthFields = (
  getString: UseStringsReturn['getString'],
  authType?: string
): ElkConnectorAuthFieldsProps[] | null => {
  if (!authType) {
    return null
  }

  switch (authType) {
    case AuthTypes.USER_PASSWORD:
      return [
        { name: ELKConnectorFields.USERNAME, label: getString('username'), key: ELKConnectorFields.USERNAME },
        { name: ELKConnectorFields.PASSWORD, label: getString('password'), key: ELKConnectorFields.PASSWORD }
      ]

    case AuthTypes.API_CLIENT_TOKEN:
      return [
        {
          name: ELKConnectorFields.API_KEY_ID,
          label: getString('platform.connectors.elk.apiId'),
          key: ELKConnectorFields.API_KEY_ID
        },
        {
          name: ELKConnectorFields.API_KEY_REF,
          label: getString('common.apikey'),
          key: ELKConnectorFields.API_KEY_REF
        }
      ]

    case AuthTypes.BEARER_TOKEN:
      return [
        {
          name: ELKConnectorFields.API_KEY_REF,
          label: getString('platform.connectors.bearerToken'),
          key: ELKConnectorFields.API_KEY_REF
        }
      ]

    default:
      return null
  }
}
