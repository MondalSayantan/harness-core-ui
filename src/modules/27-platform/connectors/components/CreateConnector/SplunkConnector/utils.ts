/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormData } from '@platform/connectors/interfaces/ConnectorInterface'
import { AuthTypes } from '@platform/connectors/pages/connectors/utils/ConnectorHelper'
import type { ConnectorConfigDTO } from 'services/cd-ng'

type InitializeSplunkConnectorArgs = {
  prevStepData?: ConnectorConfigDTO
  accountId: string
  projectIdentifier?: string
  orgIdentifier?: string
}

export function initializeSplunkConnector({
  prevStepData,
  projectIdentifier,
  accountId,
  orgIdentifier
}: InitializeSplunkConnectorArgs): FormData {
  const defaultObj = {
    url: '',
    authType: AuthTypes.USER_PASSWORD,
    username: '',
    passwordRef: undefined,
    tokenRef: undefined,
    accountId: accountId,
    projectIdentifier,
    orgIdentifier
  }

  if (!prevStepData) {
    return defaultObj
  }

  const { spec, ...prevData } = prevStepData
  return {
    ...defaultObj,
    url: prevData?.url || spec?.splunkUrl || '',
    username: prevData?.username || spec?.username,
    passwordRef: prevData?.passwordRef || spec?.passwordRef,
    authType: prevData?.authType || spec?.type || defaultObj.authType,
    tokenRef: prevData?.tokenRef || spec?.tokenRef
  }
}
