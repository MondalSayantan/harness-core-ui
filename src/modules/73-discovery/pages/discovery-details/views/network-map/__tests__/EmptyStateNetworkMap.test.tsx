/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { TestWrapper } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import { accountPathProps, discoveryPathProps, modulePathProps, projectPathProps } from '@common/utils/routeUtils'
import EmptyStateNetworkMap from '../EmptyStateNetworkMap'

const PATH = routes.toDiscoveredResource({
  ...accountPathProps,
  ...projectPathProps,
  ...modulePathProps,
  ...discoveryPathProps
})
const PATH_PARAMS = {
  accountId: 'accountId',
  orgIdentifier: 'default',
  projectIdentifier: 'Discovery_Test',
  module: 'chaos',
  dAgentId: 'dAgentId-1'
}

describe('<DiscoveryHistory /> tests', () => {
  test('should match snapshot', async () => {
    const { container } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <EmptyStateNetworkMap />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('click on create new network map', async () => {
    const { container, getByText, getByTestId } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <EmptyStateNetworkMap />
      </TestWrapper>
    )

    const newNetworkMapBtn = getByText('discovery.createNewNetworkMap')
    act(() => {
      fireEvent.click(newNetworkMapBtn)
    })

    expect(getByTestId('location')).toMatchInlineSnapshot(`
      <div
        data-testid="location"
      >
        /account/accountId/chaos/orgs/default/projects/Discovery_Test/setup/resources/discovery/dAgentId-1/network-map-studio/-1
      </div>
    `)

    expect(container).toMatchSnapshot()
  })
})
