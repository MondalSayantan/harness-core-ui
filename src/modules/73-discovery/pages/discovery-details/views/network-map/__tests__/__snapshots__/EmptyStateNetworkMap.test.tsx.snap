// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscoveryHistory /> tests click on create new network map 2`] = `
<div>
  <div>
    <h1>
      Not Found
    </h1>
    <div
      data-testid="location"
    >
      /account/accountId/chaos/orgs/default/projects/Discovery_Test/setup/resources/discovery/dAgentId-1/network-map-studio/-1
    </div>
  </div>
</div>
`;

exports[`<DiscoveryHistory /> tests should match snapshot 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main noNetworkMapDiv"
    style="width: 80%;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-xxxlarge StyledProps--main"
      style="margin: auto; padding-right: 20px; width: 60%;"
    >
      <img
        src="test-file-stub"
        width="140px"
      />
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-h2"
      >
        discovery.discoveryDetails.networkMaps.noNetworkMapHeader
      </p>
      <p
        class="StyledProps--font StyledProps--main"
      >
        discovery.discoveryDetails.networkMaps.noNetworkMapDesc
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
        style="width: 350px;"
      >
        <div
          style="display: flex; align-items: center;"
        >
          <div
            class="iconDiv"
          >
            <span
              class="bp3-icon StyledProps--main moduleIcons"
              data-icon="code-settings"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 13 13"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.477 1.705c0-.27.218-.488.487-.488h.975c.27 0 .488.218.488.488v1.138c.31.084.606.206.879.363l.81-.807c.19-.189.498-.189.688 0l.69.688c.192.19.192.5 0 .691l-.807.804c.156.27.278.563.362.87h1.144c.269 0 .487.22.487.488v.975c0 .27-.218.488-.487.488h-1.144a3.687 3.687 0 01-.363.872l.807.803c.191.19.191.5 0 .691l-.69.688a.488.488 0 01-.689 0l-.81-.806a3.715 3.715 0 01-.877.362v1.138c0 .27-.219.487-.488.487h-.975a.488.488 0 01-.487-.487v-1.138a3.715 3.715 0 01-.88-.363l-.808.806a.488.488 0 01-.688 0l-.691-.688a.487.487 0 010-.69l.806-.804a3.685 3.685 0 01-.362-.871H1.71a.488.488 0 01-.487-.488V5.94c0-.269.218-.487.487-.487h1.143c.084-.308.206-.6.362-.87l-.807-.805a.488.488 0 010-.69l.69-.689c.191-.19.499-.19.689 0l.81.807c.272-.157.567-.28.879-.363zm.65.162v1.1a.493.493 0 01-.383.478 3.065 3.065 0 00-.911.377.493.493 0 01-.607-.069l-.782-.779-.46.459.779.776a.493.493 0 01.069.61c-.171.275-.3.58-.376.903a.493.493 0 01-.479.38H1.873v.65h1.104c.237 0 .428.167.479.382.077.324.205.628.376.903.117.188.1.443-.069.61l-.779.776.46.459.783-.78a.493.493 0 01.606-.068c.278.171.585.3.911.376.216.05.383.242.383.479v1.1h.65v-1.1c0-.237.167-.429.382-.479.326-.076.633-.205.91-.375a.493.493 0 01.607.068l.782.78.46-.46-.779-.776a.493.493 0 01-.068-.61c.17-.275.3-.58.376-.903a.493.493 0 01.478-.381h1.105v-.65H9.926a.493.493 0 01-.479-.381 3.036 3.036 0 00-.376-.903.493.493 0 01.07-.61l.78-.776-.461-.459-.783.78a.493.493 0 01-.607.068 3.065 3.065 0 00-.911-.377.493.493 0 01-.382-.478v-1.1zm.385 3.09a1.544 1.544 0 100 3.087 1.544 1.544 0 000-3.088zM4.32 6.5a2.194 2.194 0 114.387 0 2.194 2.194 0 01-4.387 0z"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-left-xsmall"
          >
            discovery.discoveryDetails.networkMaps.noNetworkMapChooseService
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-arrow-right StyledProps--main"
          icon="arrow-right"
        >
          <svg
            data-icon="arrow-right"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              arrow-right
            </desc>
            <path
              d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <div
          style="display: flex; align-items: center;"
        >
          <div
            class="iconDiv"
          >
            <span
              class="bp3-icon StyledProps--main moduleIcons StyledProps--margin-top-xsmall StyledProps--margin-left-xsmall"
              data-icon="graph"
            >
              <svg
                fill="none"
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.82 6.927c-.437 0-.861.144-1.207.41L9.259 5.033a3.283 3.283 0 10-4.697 1.142l-.986 2.48a2.662 2.662 0 101.078.46l.983-2.472c.247.057.499.087.752.09.782 0 1.538-.28 2.131-.791l3.411 2.345a2.067 2.067 0 00-.102.64 1.994 1.994 0 101.99-1.993v-.007z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-left-xsmall"
          >
            discovery.tabs.configureRelations
          </p>
        </div>
      </div>
      <button
        aria-label="discovery.createNewNetworkMap"
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
        style="width: 220px;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          discovery.createNewNetworkMap
        </span>
      </button>
    </div>
    <div
      class="Layout--vertical StyledProps--main"
      style="width: 40%;"
    >
      <img
        src="test-file-stub"
        style="margin-top: 50px;"
      />
    </div>
  </div>
</div>
`;
